internal/use_cases/create_pull_request.go

```diff
@@ -43,8 +46,34 @@ type CreatePullRequest struct {
 	BranchProvider          domain.BranchProvider
 }
 
+func (cpr *CreatePullRequest) validateTemplateFile() error {
+	if cpr.Cfg.TemplatePath == "" {
+		return nil // No template to validate
+	}
+
+	templatePath := cpr.Cfg.TemplatePath
+	if !filepath.IsAbs(templatePath) {
+		repoRoot, err := cpr.Git.GetRepositoryRoot()
+		if err != nil {
+			return fmt.Errorf("failed to determine repository root: %w", err)
+		}
+		templatePath = filepath.Join(repoRoot, templatePath)
+	}
+
+	if _, err := os.Stat(templatePath); os.IsNotExist(err) {
+		return fmt.Errorf("template file does not exist: %s", cpr.Cfg.TemplatePath)
+	}
```

Comment:
suggestion: return error in case error is other than not exists. `os.Stat` can return another error (such as permission).
```suggestion
	if _, err := os.Stat(templatePath) {
		if errors.Is(err, fs.ErrNotExist) {
			return fmt.Errorf("template file does not exist: %s", cpr.Cfg.TemplatePath)
		}
		
		// Fallback in case is other error. We can format a message here too.
		return err
	}
```